### PR TITLE
Fix KDE Global theme names

### LINKED ISSUE #23: Theme names don't fit
When selecting KDE themes on the settings option, some of the larger names don't fit the box. 
example

### LINKED ISSUE #30: Impossible to read config values for KDE themes (color schemes?)
Widget too small to display content:

image

### PR DESCRIPTION
**Pull request by l0drex**
I try to display more beautiful theme names for Global KDE Themes in the settings.

Fixes #23
Fixes #30

### CODE REVIEW

**Comment by l0drex** [comment]
I'm now at a point where I have the real names and the ones used earlier. I guess I have to delete a specific file that is generated for the UI?

**Comment by daehruoydeef** [comment]
Looks good so far and a smart move to take the original theme names! But we are facing some problems here, a quick overview how it currently works:

1) Settings UI is loaded
2) `get_kde_themes()` is called and updates the options(light and dark) for the combofields (Theme Dropdowns) and adds in the new values. Please note that at this point the Dropdown will not default to a theme since no theme is selected at this point, only the dropdown options are added.
3) [In this line](https://github.com/daehruoydeef/Yin-Yang/blob/f47188f1302777a700937a3a89560deb41d7978e/src/gui.py#L80) we set the default for the selected option in the combofield. So we go through the options added in the step before and see if one matches with the one set inside the config. We save the index of the found option
4) Now we set the index of the found theme as the currently selected option.

This is currently how this is handled. One more problem will occur when we are going to save. When we are saving it will use the currently selected theme and update this inside the config file.
So if we are using the short theme names it will update this inside the config and later on break any functionality that needs the full length name like `lookandfeeltool`


#### What could be a possible solution?
Basically we need to map the full length theme names with the short names and use the short names inside the dropdown but save the long names. Before the default option is selected we simply map it to the short name and set this instead.


I did some research on my own system and found something that could break it, so error handling is needed. Basically check the screenshot -> so if no theme name is found it should use the full length name instead.
image

**Comment by l0drex** [comment]
I created now 3 functions:

1. get_kde_theme_names() returns a map with short names as key and long names as value. If no short name was found, the key is the long name.
2. get_kde_theme_short() translates a long theme name to a short theme name,
3. get_kde_theme_long() does the opposite.

It seems to work, but I still can't figure out why I have both theme names now in the dropdown:
grafik

**Comment by daehruoydeef** [comment]
There was a small error inside the if condition, since we are dealing with a map we cannot use `in` without providing a key/value pair. 

I added a simple exception to catch 2 errors. One error was that it will crash if no file is found, I also used this exception to say if no file is found just use the long name. Check the Code below. It works fine with that.

```
        # get the actual name
        for long in long_names:
            # trying to get the Desktop file
            try:
                # load the name from the metadata.desktop file
                with open('/usr/share/plasma/look-and-feel/{long}/metadata.desktop'.format(**locals()), 'r') as file:
                    # search for the name
                    for line in file:
                        if 'Name=' in line:
                            name: str = ''
                            write: bool = False
                            for letter in line:
                                if letter == '\n':
                                    write = False
                                if write:
                                    name += letter
                                if letter == '=':
                                    write = True
                            print(name)
                            themes[name] = long
                            break
            # if no file exist lets just use the long name
            except: 
                themes[long] = long
        return themes
```

**Comment by l0drex** [comment]
Oh, ok, thanks :)
Works flawlessly now.

**Comment by tneo** [comment]
I ran git pull and install again, but I don't see this change in my Yin-Yang. Did I do something wrong?
image

**Comment by daehruoydeef** [comment]
No you did everything correct, the problem here lies that every theme creator installs the theme differently. You can read the thread to see how it is currently implemented. 

If you know where on your system the themes with the long names are installed please let me know

**Comment by tneo** [comment]
The location is ~/.local/share/plasma/look-and-feel/ and the drop down seems to be using the directory names and not what is given with the name tag in the .desktop file for a theme:
image

**Comment by daehruoydeef** [comment]
Yes this is already used. However there are some themes that are installed differently. There would also need to be a function in places that takes the long name and breaks it down to the last part.

**Comment by tneo** [comment]
I think you use the X-KDE-PluginInfo-Name value in the metadata.desktop file instead of the Name field.

For instance for the Inverse theme:
`Name=Inverse
X-KDE-PluginInfo-Name=com.github.yeyushengfan258.Inverse`

**Comment by daehruoydeef** [comment]
@tneo @l0drex this is fixed and on master now. Theme names are working fine now see
#49

**Comment by tneo** [comment]
Nice! :-)
Can you sort the list alphabetically in the drop-down, so that is easier to read?
